src/main/java/org/opensearch/knn/index/codec/util/NativeMemoryCacheKeyHelper.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.util;
+
+import org.apache.lucene.index.SegmentInfo;
+
+import java.util.Base64;
+
+public final class NativeMemoryCacheKeyHelper {
+    private NativeMemoryCacheKeyHelper() {}
+
+    /**
+     * Construct a unique cache key for look-up operation in {@link org.opensearch.knn.index.memory.NativeMemoryCacheManager}
+     *
+     * @param vectorIndexFileName Vector index file name. Ex: _0_165_test_field.faiss.
+     * @param segmentInfo Segment info object representing a logical segment unit containing a vector index.
+     * @return Unique cache key that can be used for look-up and invalidating in
+     * {@link org.opensearch.knn.index.memory.NativeMemoryCacheManager}
+     */
+    public static String constructCacheKey(final String vectorIndexFileName, final SegmentInfo segmentInfo) {
+        final String segmentId = Base64.getEncoder().encodeToString(segmentInfo.getId());
```

Comment:
Since the cache manager is being used across all indices, merely having index file name would lead to a conflict.
For example, more than one index can have `_0_165_my_field.faiss` as a vector index file, while each index suppose to have a different data.

When we were using a fullpath as a cache key, which has not only the index hash but also it had shard id, prevented this from happening, but after we cut out FSDirectory dependency, we need to add a uniqueness to the file name as a cache key.

Spent quite a long time to find a way to pull index hash + shard id into places they need, I concluded just using a segment unique id would be simple to implement.

Reply:
once we have this as the cache key, what is the impact on the KNNGraph stats? can we still see how many graph files are loaded per Opensearch index?

Reply:
since we don't care the actual values in the map, but we do care the aggregated level statistics, it should be fine!
```
    public Map<String, Map<String, Object>> getIndicesCacheStats() {
        Map<String, Map<String, Object>> statValues = new HashMap<>();
        NativeMemoryAllocation.IndexAllocation indexAllocation;

        for (Map.Entry<String, NativeMemoryAllocation> entry : cache.asMap().entrySet()) {

            if (entry.getValue() instanceof NativeMemoryAllocation.IndexAllocation) {
                indexAllocation = (NativeMemoryAllocation.IndexAllocation) entry.getValue();
                String indexName = indexAllocation.getOpenSearchIndexName();

                Map<String, Object> indexMap = statValues.computeIfAbsent(indexName, name -> new HashMap<>());

                vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
                indexMap.computeIfAbsent(GRAPH_COUNT, key -> getIndexGraphCount(indexName));
                indexMap.computeIfAbsent(StatNames.GRAPH_MEMORY_USAGE.getName(), key -> getIndexSizeInKilobytes(indexName));
                indexMap.computeIfAbsent(StatNames.GRAPH_MEMORY_USAGE_PERCENTAGE.getName(), key -> getIndexSizeAsPercentage(indexName));
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            }
        }

        return statValues;
    }
```
we don't care about the actual values. what we care is the statistics like total #graph loaded etc.
and in my change, I changed the path (which was used as a cache key) to `<file name>@<segment id>` format, so the stats remain same.